entrypoint.sh

```diff
@@ -8,26 +8,34 @@ APP_GROUP=steam
 APP_HOME=/home/$APP_USER
 
 source /includes/colors.sh
+source /includes/permissions.sh
 
 if [[ "${PUID}" -eq 0 ]] || [[ "${PGID}" -eq 0 ]]; then
-    ee ">>> Running as root is not supported, please fix your PUID and PGID!"
+    ee ">>> Running Palworld as root is not supported, please fix your PUID and PGID!"
     exit 1
 elif [[ "$(id -u steam)" -ne "${PUID}" ]] || [[ "$(id -g steam)" -ne "${PGID}" ]]; then
     ew "> Current $APP_USER user PUID is '$(id -u steam)' and PGID is '$(id -g steam)'"
     ew "> Setting new $APP_USER user PUID to '${PUID}' and PGID to '${PGID}'"
     groupmod -g "${PGID}" "$APP_GROUP" && usermod -u "${PUID}" -g "${PGID}" "$APP_USER"
-else 
+else
     ew "> Current $APP_USER user PUID is '$(id -u steam)' and PGID is '$(id -g steam)'"
     ew "> PUID and PGID matching what is requested for user $APP_USER"
 fi
 
-chown -R "$APP_USER":"$APP_GROUP" "$APP_HOME"
-chown -R "$APP_USER":"$APP_GROUP" "$GAME_ROOT"
-chown "$APP_USER":"$APP_GROUP" /entrypoint.sh
-chown "$APP_USER":"$APP_GROUP" /PalWorldSettings.ini.template
-chown -R "$APP_USER":"$APP_GROUP" /scripts
-chown -R "$APP_USER":"$APP_GROUP" /includes
+change_ownership $APP_USER $APP_GROUP "$APP_HOME"
```

Comment:
Why do we need an alias function for that?
Please explain it

Reply:
The function allows us to only attempt to change the file permissions when they are not set properly and we are running as root. By only attempting when they are not properly set, we allow more advanced users to run the container as a non-root user as requested in https://github.com/jammsen/docker-palworld-dedicated-server/issues/203#issuecomment-1948244499

By only attempting to change the permissions when running as root, it allows us to provide a user friendly error message. This is to help people who are confused by the "Operation not permitted" error message of #203. 

I'll add the following comments to the actual file for future reference, not on the right computer right now. 

Changes ownership of the given target to the given user:group. Can accept a file or folder as a target. Will be recursive if a folder. Prior to changing the permissions, it first checks if the permissions are already set appropriately. If they are, no operation is performed. If they are not, then it will check if running as root. If not, it will exit with an error message. If it is running as root, it will update the permissions.

Reply:
> I'll add the following comments to the actual file for future reference, not on the right computer right now.
> 
> Changes ownership of the given target to the given user:group. Can accept a file or folder as a target. Will be recursive if a folder. Prior to changing the permissions, it first checks if the permissions are already set appropriately. If they are, no operation is performed. If they are not, then it will check if running as root. If not, it will exit with an error message. If it is running as root, it will update the permissions.

Ah i think i understand now more of you viewpoint, you want to make sure to not do un-needed work by checking file permissions first and than generate a list to give permissions in a way when the IDs dont change. Thats where we differ, i expect the IDs to change every start, like a random number generator and make sure that "EVERY" time the permissions are right outright, without the check because i know what im looking for and what is needed. I outright refuse the use of root here, because of the problems in permissions it brings with. Docker-Containers should be stateless and make sure they can work as they need it. Thats why im doing what im doing.

> The function allows us to only attempt to change the file permissions when they are not set properly and we are running as root. By only attempting when they are not properly set, we allow more advanced users to run the container as a non-root user as requested in [#203 (comment)](https://github.com/jammsen/docker-palworld-dedicated-server/issues/203#issuecomment-1948244499)
> 
> By only attempting to change the permissions when running as root, it allows us to provide a user friendly error message. This is to help people who are confused by the "Operation not permitted" error message of #203.

See this is i think where the perception is totally wrong, im using 3 setups to test this: 
1.) Windows11 + WSL2 and Docker Desktop under my user 
  - So im not root as the user who wants to the container, but the service is still running as root-ful because the software is run inside HyperV over Docker-Desktop
  - I can switch that by using user: ID:ID in Docker-Compose in my test-docker-compose-files
2.) Ubuntu VM with Docker-Node for testing, but Docker root-less
  - Have my non-root user and non-root-ful Docker and it works as expected
3.) Debian 11 VM with Docker 25 under only the user root
  - I can run all as root and i created a non-login user with an id for steam usage with a certain UID/GID

In all of the setups i can run this and it works, because it respects what im trying to do, EXCEPT running the gameserver as only root and that is all by design.

Regarding the exact comment on issue 203 i can only recommend to ignore it, the comment is full of contradicting informations, mixing topics in a way that looks good to enable scare-talk, which in-itself when you have works with Docker longer then 3 weeks totally falls apart because it doesnt make sense ... on a highlevel fly-over as a new-user i sounds almost logical and dangerous to listen too, but thats all that it is, is miss-information to enable scare-talk to setup the user into dummy mode and almost tell him, go download my script and run it as root "Trust me. Bro!". That degree of dangerous miss-information is no joke, thats why i wrote this:

https://github.com/jammsen/docker-palworld-dedicated-server/issues/203#issuecomment-1949077706

Also, just to clarify, most of the people who acutally had problems after the update were Portainer users, who have to deal with the constant bending and breaking of rules, which do not exist in Docker-Compose or Docker. But the problem was that most of the users dont understand how and why Portainer does things and where to look for in changes of the Dockerfile and transfer it into the weird config of Portainer. From a Repo-POV which solves this with Docker-Compose this could be considered an edge-case.